OreDict.body.php

```diff
@@ -59,26 +55,21 @@ public function getRawOutput() {
 
 	/**
 	 * Execute output hooks and returns their output.
-	 *
-	 * @param string $params
-	 * @return array
 	 */
 
-	public function runHooks($params = "") {
+	public function runHooks($params = ""): array {
+		$hooks = MediaWikiServices::getInstance()->getHookContainer();
```

Comment:
maybe we could move this to constructor?

Reply:
This class is weird and I don't want to mess with that this time. It has multiple roles, so If I were to refactor it I would probably start with that, and then a service that can have injected dependencies would probably emerge as part of it. As is, the class constructor is being called in contexts that shouldn't need to depend on service container, so it would just be pushing the problem around